aio_pika/channel.py

```diff
@@ -56,158 +55,180 @@ def __init__(
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         conn = None
 
-        if self._channel:
+        if self.is_opened:
             conn = self._channel.connection
 
-        return '<%s "%s#%s">' % (self.__class__.__name__, conn, self)
-
-    def __iter__(self):
-        return (yield from self.__await__())
+        return '<%s #%s "%s">' % (self.__class__.__name__, self, conn)
 
-    def __await__(self):
+    def __await__(self) -> Generator[Any, Any, "AbstractChannel"]:
         yield from self.initialize().__await__()
         return self
 
-    async def __aenter__(self):
-        if not self._channel:
+    async def __aenter__(self) -> "AbstractChannel":
+        if not hasattr(self, "_channel"):
             await self.initialize()
-
         return self
 
-    async def __aexit__(self, exc_type, exc_val, exc_tb):
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
         await self.close()
 
     def add_close_callback(
-        self, callback: CloseCallbackType, weak: bool = False
+        self, callback: ChannelCloseCallback, weak: bool = False,
     ) -> None:
-        self._done_callbacks.add(callback, weak=weak)
+        self.close_callbacks.add(callback, weak=weak)
 
     def remove_close_callback(
-        self, callback: CloseCallbackType
+        self, callback: ChannelCloseCallback,
     ) -> None:
-        self._done_callbacks.remove(callback)
+        self.close_callbacks.remove(callback)
 
     def add_on_return_callback(
-        self, callback: ReturnCallbackType, weak: bool = False,
+        self, callback: ReturnCallback, weak: bool = False,
     ) -> None:
-        self._return_callbacks.add(callback, weak=weak)
+        self.return_callbacks.add(callback, weak=weak)
 
-    def remove_on_return_callback(
-        self, callback: ReturnCallbackType, weak: bool = False
-    ) -> None:
-        self._return_callbacks.remove(callback)
+    def remove_on_return_callback(self, callback: ReturnCallback) -> None:
+        self.return_callbacks.remove(callback)
 
-    async def _create_channel(self) -> aiormq.Channel:
-        await self._connection.ready()
+    async def _create_channel(
+        self, timeout: TimeoutType = None,
+    ) -> aiormq.abc.AbstractChannel:
+        await self.connection.ready()
 
-        return await self._connection.connection.channel(
+        return await self.connection.connection.channel(
             publisher_confirms=self._publisher_confirms,
             on_return_raises=self._on_return_raises,
             channel_number=self._channel_number,
+            timeout=timeout,
         )
 
     async def initialize(self, timeout: TimeoutType = None) -> None:
-        if self._channel is not None:
-            raise RuntimeError("Can't initialize channel")
+        if self.is_opened:
```

Comment:
Maybe add some verbolize in is_close and use only that? 
Always checks for `is_opened` and `_is_closed_by_user` looks verbose and very easy to forget one of this

Reply:
It's semantically different cases. `is_opened` means the channel is ready for interaction. `is_closed` means the channel has been closed.